### ISSUE TITLE
Generalize `pudl.extract.raw_df_factory` to extract any partition (not just `year`)

### ISSUE THREAD

**Issue by cmgosnell** [catalyst-cooperative/pudl#3366]
right now `pudl.extract.raw_df_factory` expects the partition of the extractor to be `year` which is hard coded in a few places.

See [comment](https://github.com/catalyst-cooperative/pudl/pull/3331#discussion_r1478958501) which spurred this issue.

```[tasklist]
- [ ] generalize `year_extractor_factory` -> `partition_extractor_factory`
- [ ] generalize `extract_single_year` -> `extract_single_partition`
- [ ] generalize `years_from_settings_factory` -> `partitions_from_settings_factory`
- [ ] integrate partition changes into `raw_df_factory` & `raw_df_factory.raw_dfs`
- [ ] test new setup with existing year-based use of excel extractor (eia860, eia923, phmsagas.... (i think that's all?)
- [ ] convert `pudl.extract.eia860m.raw_eia860m__all_dfs` to use `raw_df_factory`
- [ ] ...(i'm sure i missed something)?
```

**Comment by NateWasTaken** [comment]
Hello can I take this assignment?

**Comment by NateWasTaken** [comment]
Sorry - forgot to tag. @catalyst-cooperative/com-dev Can I take this assignment

**Comment by NateWasTaken** [comment]
> Hi @NateWasTaken! Thanks for the message. This issue has actually been addressed in #3402 and should be closed, my mistake! I'll take a look through our existing issues and see what other candidates for good first issues I can find.

That would be great thank you e-belfer. My partner and I are working on a class project and hoping to contribute in any way possible!
